test/spec/oidc/verifyToken.ts

```diff
@@ -15,11 +62,147 @@ describe('token.verify', function() {
   var validationParams;
   var client;
   beforeEach(() => {
+    mocked.decodeToken.mockImplementation(original.decodeToken);
+    mocked.validateClaims.mockImplementation(original.validateClaims);
+    mocked.getWellKnown.mockReturnValue(Promise.resolve({
+      issuer: tokens.standardIdTokenParsed.issuer
+    }));
+    mocked.getKey.mockReturnValue(Promise.resolve('fake-test-key'));
+    mocked.verifyToken.mockReturnValue(Promise.resolve(true));
+    util.warpToUnixTime(1449699929);
     validationParams = {
       clientId: tokens.standardIdTokenParsed.clientId,
       issuer: tokens.standardIdTokenParsed.issuer
     };
     client = setupSync();
+    oauthUtil.loadWellKnownAndKeysCache(client);
+  });
+  afterEach(() => {
+    Object.keys(mocked).forEach(key => {
+      mocked[key].mockReset();
+    });
+  });
+  describe('decodeToken', () => {
+    it('passes the idToken string to decodeToken', async () => {
+      await client.token.verify(tokens.standardIdTokenParsed, validationParams);
+      expect(mocked.decodeToken).toHaveBeenCalledWith(tokens.standardIdTokenParsed.idToken);
+    });
+    it('will throw errors from decodeToken', async () => {
+      const error = new Error('a fake test error');
+      mocked.decodeToken.mockImplementation(() => {
+        throw error;
+      });
+      await expect(client.token.verify(tokens.standardIdTokenParsed, validationParams)).rejects.toThrow(error);
+    });
+  });
+  describe('validationOptions', () => {
+    describe('defaults', () => {
+      it('issuer: will use well-known endpoint', async () => {
+        client = setupSync({
+          issuer: 'http://some-proxy'
+        });
+        await client.token.verify(tokens.standardIdTokenParsed, { clientId: tokens.standardIdTokenParsed.clientId });
+        expect(mocked.validateClaims.mock.calls[0][2].issuer).toBe(tokens.standardIdTokenParsed.issuer);
+      });
+      it('clientId: will use sdk option', async () => {
+        client = setupSync({
+          clientId: tokens.standardIdTokenParsed.clientId,
+        });
+        await client.token.verify(tokens.standardIdTokenParsed, undefined);
+        expect(mocked.validateClaims.mock.calls[0][2].clientId).toBe(tokens.standardIdTokenParsed.clientId);
+      });
+      it('ignoreSignature: will use sdk option', async () => {
+        client = setupSync({
+          clientId: tokens.standardIdTokenParsed.clientId,
+          ignoreSignature: true
+        });
+        await client.token.verify(tokens.standardIdTokenParsed, undefined);
+        expect(mocked.validateClaims.mock.calls[0][2].ignoreSignature).toBe(true);
+      });
+    });
+    it('can override all defaults by passing validationParams', async () => {
```

Comment:
I don't see `validationParams` mentioned anywhere in the README, did I miss something?

Reply:
https://github.com/okta/okta-auth-js#tokenverifyidtokenobject The 2nd argument to the function is called `validationOptions` in the README.  In t[he code,](https://github.com/okta/okta-auth-js/pull/646/files/74030838134939d728f106167ee4b83dbdca87f5#diff-d1a78922dd3b1054ff328c8c19058be489a5fb17a2a4067461197bf00fda6d23) validationOptions is the final object made from combining the user passed options with the defaults. validationParams refers only to the user passed options

Reply:
Okay, I searched for the wrong string, this is fine.